mslice/plotting/plot_window/plot_figure.py

```diff
@@ -65,74 +73,50 @@ def _get_min(self, data, absolute_minimum=-np.inf):
                 pass
         return np.min(running_min) if running_min else absolute_minimum
 
-    def _apply_config(self, plot_config): # noqa: C901
+    def change_cut_plot(self, xy_config):
         current_axis = self.canvas.figure.gca()
-        current_axis.set_title(plot_config.title)
-        current_axis.set_xlabel(plot_config.xlabel)
-        current_axis.set_ylabel(plot_config.ylabel)
-
-        if current_axis.get_images():
-            images = current_axis.get_images()
-            if len(images) != 1:
-                raise RuntimeError("Expected single image on axes, found " + str(len(images)))
-            mappable = images[0]
-            mappable.set_clim(*plot_config.colorbar_range)
-            vmin, vmax = plot_config.colorbar_range
-
-            if plot_config.logarithmic and type(mappable.norm) != colors.LogNorm:
-                mappable.colorbar.remove()
-                if vmin == float(0):
-                    vmin = 0.001
-                norm = colors.LogNorm(vmin, vmax)
-                mappable.set_norm(norm)
-                self.canvas.figure.colorbar(mappable)
-            elif not plot_config.logarithmic and type(mappable.norm) != colors.Normalize:
-                mappable.colorbar.remove()
-                norm = colors.Normalize(vmin, vmax)
-                mappable.set_norm(norm)
-                self.canvas.figure.colorbar(mappable)
+        if xy_config['x_log']:
+            xdata = [ll.get_xdata() for ll in current_axis.get_lines()]
+            xmin = self.get_min(xdata, absolute_minimum=0.)
+            current_axis.set_xscale('symlog', linthreshx=pow(10, np.floor(np.log10(xmin))))
+            if self.get_min(xdata) > 0:
```

Comment:
Is there a reason to do this calculation again rather than just using the value calculated 2 lines above?